src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/compute/v2/OpenStackComputePluginTest.java

```diff
@@ -1,858 +1,688 @@
-                "default", null, bestCpu, bestMemory, bestDisk, imageId, null, publicKey, networksId);
-        JSONObject computeJson = generateJsonRequest(imageId, bestFlavorId, userData, idKeyName, responseNetworkIds,
-                idInstanceName);
-
-        mockGetFlavorsRequest(bestFlavorId, bestCpu, bestMemory, bestDisk);
-        Mockito.when(this.clientMock.doPostRequest(this.argString.capture(), this.argBodyString.capture(), this.argCloudToken.capture()
-        )).thenReturn("");
-        Mockito.when(this.launchCommandGeneratorMock.createLaunchCommand(computeOrder)).thenReturn(userData);
-        Mockito.doReturn(idKeyName).doReturn(idInstanceName).when(this.computePlugin).getRandomUUID();
-        Mockito.doReturn(expectedInstanceIdJson).when(this.clientMock).doPostRequest(argString.capture(),
-                argBodyString.capture(), argCloudToken.capture());
-
-        //exercise 1
-        String instanceId = this.computePlugin.requestInstance(computeOrder, this.openStackV3User);
-
-        //verify 1
-        Assert.assertEquals(this.argString.getAllValues().get(0), this.osKeyPairEndpoint);
-        Assert.assertEquals(this.argCloudToken.getAllValues().get(0), this.openStackV3User);
-        JSONAssert.assertEquals(this.argBodyString.getAllValues().get(0).toString(), rootKeypairJson.toString(), false);
-
-        Assert.assertEquals(this.argString.getAllValues().get(1), computeEndpoint);
-        Assert.assertEquals(this.argCloudToken.getAllValues().get(1), this.openStackV3User);
-        // ToDo: fix the assertion below
-        //JSONAssert.assertEquals(this.argBodyString.getAllValues().get(1), computeJson.toString(), false);
-        Assert.assertEquals(expectedInstanceId, instanceId);
-
-        // exercise 2
-        instanceId = this.computePlugin.requestInstance(computeOrder, this.openStackV3User);
-
-        // verify 2
-        Assert.assertEquals(this.argString.getAllValues().get(0), this.osKeyPairEndpoint);
-        Assert.assertEquals(this.argCloudToken.getAllValues().get(0), this.openStackV3User);
-        JSONAssert.assertEquals(this.argBodyString.getAllValues().get(0).toString(), rootKeypairJson.toString(), false);
-
-        Assert.assertEquals(this.argString.getAllValues().get(1), computeEndpoint);
-        Assert.assertEquals(this.argCloudToken.getAllValues().get(1), this.openStackV3User);
-        // ToDo: fix the assertion below
-        // JSONAssert.assertEquals(this.argBodyString.getAllValues().get(1).toString(), computeJson.toString(), false);
-        Assert.assertEquals(expectedInstanceId, instanceId);
-    }
-
-    // test case: Get Instance should still work even if there is no address field on response
+    // test case: checks if a given flavor has a requirement
     @Test
-    public void testGetInstanceWhenThereIsNoAddressFieldOnResponse() throws FogbowException, HttpResponseException {
+    public void testFlavorHasRequirements() throws FogbowException, HttpResponseException {
```

Comment:
there are plenty others methods you should verify if have been called. Take a look at that please.